src/components/UserSelector.tsx

```diff
@@ -1,19 +1,69 @@
-import React from 'react';
+import React, {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type Props = {
+  users: User[] | null | undefined
+  setDropDownOpen: Dispatch<SetStateAction<boolean>>;
+  dropDownOpen: boolean;
+  setUserSelectedId: Dispatch<SetStateAction<number>>;
+  userSelectedId: number;
+  setPostSelectedId: Dispatch<SetStateAction<number>>
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  setDropDownOpen,
+  dropDownOpen,
+  setUserSelectedId,
+  userSelectedId,
+  setPostSelectedId,
+}) => {
+  const [selectedUser, setSelectedUser] = useState<User>();
+  const showOptions = () => {
+    setDropDownOpen(prevState => !prevState);
+  };
+
+  const handleUserKeyPress = useCallback(event => {
+    const dropDown = document.getElementById('dropdown-menu');
+
+    if (dropDown && !dropDown.contains(event.target)) {
+      setDropDownOpen(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener('click', handleUserKeyPress);
+
+    return () => {
+      window.removeEventListener('click', handleUserKeyPress);
+    };
+  }, [handleUserKeyPress]);
 
-export const UserSelector: React.FC = () => {
   return (
     <div
       data-cy="UserSelector"
-      className="dropdown is-active"
+      id="dropdown-menu"
+      className={classNames('dropdown', {
+        'is-active': dropDownOpen,
+      })}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={showOptions}
         >
-          <span>Choose a user</span>
+          {!selectedUser ? <span>Choose a user</span>
+            : <span>{selectedUser.name}</span>}
```

Comment:
Doesn't look good. Try to think about the readability 
```suggestion
          {!selectedUser 
            ? <span>Choose a user</span>
            : <span>{selectedUser.name}</span>}
```

Reply:
Done